gensim/models/word2vec.py

```diff
@@ -508,130 +233,59 @@ def __init__(self, sentences=None, size=100, alpha=0.025, window=5, min_count=5,
 
         """
 
+        self.callbacks = callbacks
         self.load = call_on_class_only
 
         if FAST_VERSION == -1:
             logger.warning('Slow version of %s is being used', __name__)
         else:
             logger.debug('Fast version of %s is being used', __name__)
 
-        self.initialize_word_vectors()
         self.sg = int(sg)
-        self.cum_table = None  # for negative sampling
-        self.vector_size = int(size)
-        self.layer1_size = int(size)
         if size % 4 != 0:
             logger.warning("consider setting layer size to a multiple of 4 for greater performance")
         self.alpha = float(alpha)
-        self.min_alpha_yet_reached = float(alpha)  # To warn user if alpha increases
         self.window = int(window)
-        self.max_vocab_size = max_vocab_size
-        self.seed = seed
         self.random = random.RandomState(seed)
-        self.min_count = min_count
-        self.sample = sample
-        self.workers = int(workers)
         self.min_alpha = float(min_alpha)
         self.hs = hs
         self.negative = negative
         self.cbow_mean = int(cbow_mean)
-        self.hashfxn = hashfxn
-        self.iter = iter
-        self.null_word = null_word
-        self.train_count = 0
-        self.total_train_time = 0
-        self.sorted_vocab = sorted_vocab
-        self.batch_words = batch_words
-        self.model_trimmed_post_training = False
+        # self.null_word = null_word
         self.compute_loss = compute_loss
         self.running_training_loss = 0
+
+        self.wv = Word2VecKeyedVectors()
+        self.vocabulary = Word2VecVocab(
+            max_vocab_size=max_vocab_size, min_count=min_count, sample=sample,
+            sorted_vocab=sorted_vocab, null_word=null_word)
```

Comment:
`sorted_vocab=bool(sorted_vocab)` better since the param is a boolean in `Word2VecVocab`.